pkg/docker/mover.go

```diff
@@ -0,0 +1,50 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+)
+
+type DockerClient interface {
+	PullImage(ctx context.Context, image string) error
+	PushImage(ctx context.Context, image string, endpoint string) error
+	LoadFromFile(ctx context.Context, filepath string) error
+	SaveToFile(ctx context.Context, filepath string, images ...string) error
+}
+
+type (
+	ImageSourceLoader      func(ctx context.Context, client DockerClient, images ...string) error
+	ImageDestinationLoader func(ctx context.Context, client DockerClient, images ...string) error
+)
+
+type ImageMover struct {
+	client             DockerClient
+	loadSource         ImageSourceLoader
+	writeToDestination ImageDestinationLoader
+}
+
+func NewImageMover(client DockerClient) *ImageMover {
```

Comment:
I think there's an invariant here that's not being kept. For `ImageMover` to be useful, it _requires_ a source and destination. Therefore, any constructing initializers should accept a source and destination _or_ the `Move` call should accept them. The latter is useful if you need to delay constructing the source an destination because it separates `ImageMover` construction from the source and destination. 

I'd need the context under which this operates to help more. I suspect it can just accept them in a constructor though.